test/linter/test-descriptions.js

```diff
@@ -6,123 +6,108 @@ const { Logger } = require('./utils.js');
  */
 
 /**
- * @param {Identifier} apiData
- * @param {String} apiName
- * @param {Logger} logger
+ * Check for errors in the description of a specified statement's description and return whether there's an error and log as such
+ *
+ * @param {string} error_type The name of the error
+ * @param {string} name The name of the API method
+ * @param {Identifier} method The method's compat data
+ * @param {string} expected Expected description
+ * @param {Logger} logger The logger to output errors to
+ * @returns {void}
  */
-function hasValidConstrutorDescription(apiData, apiName, logger) {
-  const constructor = apiData[apiName];
-  if (
-    constructor &&
-    constructor.__compat.description !== `<code>${apiName}()</code> constructor`
-  ) {
-    logger.error(chalk`{red Incorrect constructor description for {bold ${apiName}()}}
-      {yellow Actual: {bold "${constructor.__compat.description || ''}"}}
-      {green Expected: {bold "<code>${apiName}()</code> constructor"}}`);
+function checkError(error_type, name, method, expected, logger) {
+  const actual = method.__compat.description || '';
+  if (actual != expected) {
+    logger.error(chalk`{red → Incorrect ${error_type} description for {bold ${name}}
+      Actual: {yellow "${actual}"}
+      Expected: {green "${expected}"}}`);
   }
 }
 
 /**
- * @param {Identifier} apiData
- * @param {String} apiName
- * @param {Logger} logger
+ * Process API data and check for any incorrect descriptions in said data, logging any errors
+ *
+ * @param {Identifier} apiData The data to test
+ * @param {string} apiName The name of the API
+ * @param {Logger} logger The logger to output errors to
+ * @returns {void}
  */
-function hasCorrectDOMEventsDescription(apiData, apiName, logger) {
-  for (const methodName in apiData) {
-    if (methodName.endsWith('_event')) {
-      const event = apiData[methodName];
-      const eventName = methodName.replace('_event', '');
-      if (event.__compat.description !== `<code>${eventName}</code> event`) {
-        logger.error(chalk`{red Incorrect event description for {bold ${apiName}#${methodName}}}
-      {yellow Actual: {bold "${event.__compat.description || ''}"}}
-      {green Expected: {bold "<code>${eventName}</code> event"}}`);
-      }
+function processApiData(apiData, apiName, logger) {
+  for (let methodName in apiData) {
```

Comment:
This can be `const`:

```suggestion
  for (const methodName in apiData) {
```